src/connector/src/sink/redis.rs

```diff
@@ -157,16 +145,38 @@ impl Sink for RedisSink {
             .filter(|(k, _)| self.pk_indices.contains(k))
             .map(|(_, v)| v.name.clone())
             .collect();
-        check_string_format(&self.config.common.key_format, &pk_set)?;
-        check_string_format(&self.config.common.value_format, &all_set)?;
+        if matches!(
+            self.format_desc.encode,
+            super::catalog::SinkEncode::Template
+        ) {
```

Comment:
Let's move `check_string_format` as part of `TemplateEncoder::new`, and just call `SinkFormatterImpl::new_with_redis` here.

Reply:
We should be calling TemplateEncoder::new multiple times during the data injection process, but we only need to check the format when creating the sink. So, it might be more appropriate to place it outside.

Reply:
It is called once per actor, and you are right we only need to check the format once. My point was about `TemplateEncoder` being self-contained. At least we can move the function definition into the `TemplateEncoder` file.

For more context, proto / avro is doing heavier work in `SinkFormatterImpl::new`, and we have opened an issue to discuss whether to optimize it from once-per-actor to only once on meta #12982